jupyter-on-gke/iap_module/iap.tf

```diff
@@ -44,19 +45,19 @@ resource "kubectl_manifest" "backend_config" {
 resource "kubectl_manifest" "managed_cert" {
   override_namespace = var.namespace
   yaml_body          = templatefile("${path.module}/deployments/managed-cert.yaml", {
-    ip_addr = "${google_compute_global_address.default.address}.nip.io"
+    ip_addr = var.url_domain_addr != "" ? var.url_domain_addr : "${google_compute_global_address.default[0].address}.nip.io"
   })
-  depends_on = [ google_compute_global_address.default, kubernetes_secret.my-secret ]
+  depends_on = [ kubernetes_secret.my-secret ]
 }
 
 # Ingress for IAP
 resource "kubectl_manifest" "static_ingress" {
   override_namespace = var.namespace
 
   yaml_body          = templatefile("${path.module}/deployments/static-ingress.yaml", {
-    static_addr_name = "${google_compute_global_address.default.name}"
+    static_addr_name = var.url_domain_addr != "" ? var.url_domain_name : "${google_compute_global_address.default[0].name}"
```

Comment:
should this be url_domain_name?

Reply:
So we don't want it to use the domain name if we are reserving an IP for them. I specified in the variable description that the domain name will only be used if they provide a domain url.